tests/unit/scripting.tcl

```diff
@@ -767,45 +767,53 @@ start_server {tags {"scripting"}} {
         }
     }
 
-#    test "allow-stale shebang flag" {
-#        r config set replica-serve-stale-data no
-#        r replicaof 127.0.0.1 1
-
-#        assert_error {MASTERDOWN Link with MASTER is down and replica-serve-stale-data is set to 'no'.} {
-#            r eval {
-#                return redis.call('get','x')
-#            } 0
-#        }
-
-#        assert_error {*'allow-stale' flag is not set on the function*} {
-#            r eval {#!lua flags=no-writes
-#                return 1m
-#            } 0
-#        }
-
-#        assert_equal [
-#            r eval {#!lua flags=allow-stale,no-writes
-#                return 1
-#            } 0
-#        ] 1
-
-
-#        assert_error {*Can not execute the command on a stale replica*} {
-#            r eval {#!lua flags=allow-stale,no-writes
-#                return redis.call('get','x')
-#            } 0
-#        }
-#        
-#        assert_match [
-#            r eval {#!lua flags=allow-stale,no-writes
-#                return redis.call('info','server')
-#            } 0
-#        ] {*redis_version*}
-#        
-
-#        r replicaof no one
-#        r config set replica-serve-stale-data yes
-#    }
+    test "allow-stale shebang flag" {
+        set sha [
+            r script load {#!lua flags=allow-stale,no-writes
+                return redis.call('info','server')
+            }
```

Comment:
let's declare this right before it is used.
i suppose the problem for which you put it here is that SCRIPT LOAD is not allowed during stale.
but i think we must fix this.
i.e. scripts are completely ephemeral, we don't care if we're connected to the master or not.

Reply:
if you agree, also update the top comment with that detail

Reply:
Wasn't sure we want to change `SCRIPT LOAD` too. But I agree. I'll fix.

Reply:
I did the fix but note that if someone does a `SCRIPT LOAD` on a stale server even without shebang annotation it'll work. I think this isn't really a breaking change.

Reply:
i think it's for the better.. i think that it aligns better with other changes in redis 7.0.
1. removing all the script propagation
2. adding ok-stale to a few other commands,.